zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/lifecycle/TimeoutLifecycleManager.java

```diff
@@ -53,7 +57,8 @@ public TimeoutLifecycleManager(ZeppelinConfiguration zConf,
     ScheduledExecutorService checkScheduler = ExecutorFactory.singleton()
         .createOrGetScheduled("TimeoutLifecycleManager", 1);
     checkScheduler.scheduleAtFixedRate(() -> {
-      if ((System.currentTimeMillis() - lastBusyTimeInMillis) > timeoutThreshold) {
+      if ((System.currentTimeMillis() - lastBusyTimeInMillis) > timeoutThreshold &&
+              noJobsRunning()) {
```

Comment:
According to the original design, if a job is running, shouldn't `lastBusyTimeInMilli` be updated?

Reply:
In one corner case, `lastBusyTimeInMilli` won't be updated, that is when zeppelin server is dead then `RemoteInterpreterServer#getProgress`  and `RemoteInterpreterServer#getStatus` won't be called. Thanks for the comment, let me update this PR with more comment to explain that.